analysis.go

```diff
@@ -50,60 +49,51 @@ type DeploymentConfigs struct {
 	} `json:"items"`
 }
 
-type CheckTask func(string, io.Writer) (Result, error)
+type CheckTask func(string, string) (Result, error)
+
+func GetAnalysisTasks(tasks chan<- Task, basepath string, projects []string, results chan<- CheckResults) {
```

Comment:
The check tasks run asynchronously, so they send their results down the results channel that is passed in here.